_posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc

```diff
@@ -0,0 +1,1934 @@
+use—and sets the variables to values appropriate for that environment. Here's an example of what
+`staging/us-east-2/stage/vpc-app/terragrunt.hcl` might look like:
+
+.infrastructure-live/staging/us-east-2/stage/vpc-app/terragrunt.hcl
+[source,hcl]
+----
+# Deploy the vpc-app module at a specific version (via the ref=xxx param)
+terraform {
+  source = "git@github.com:<ORG>/infrastructure-modules.git//networking/vpc-app?ref=v0.0.1"
+}
+
+# Set the variables for the vpc-app module in this environment
+inputs = {
+  aws_region       = "us-east-2"
+  aws_account_id   = "888888888888"
+  vpc_name         = "staging-vpc"
+  cidr_block       = "10.10.0.0/16"
+  num_nat_gateways = 1
+}
+
+# Automatically include settings from the root terragrunt.hcl in this account
+include {
+  path = find_in_parent_folders()
+}
+----
+
+To deploy `vpc-app` in staging, you do the following:
+
+----
+cd infrastructure-live/staging/us-east-2/stage/vpc-app
+terragrunt apply
+----
+
+When you run this command, Terragrunt will:
+
+. Checkout the `infrastructure-modules` repo at version `v0.0.1` into a scratch directory.
+. Run `terraform init` in the scratch directory, configuring the backend to the values in the root `terragrunt.hcl`.
+. Run `terraform apply` in the scratch directory, configuring the variables to the values in the `inputs = { ... }`
+  block.
+
+You can deploy the production environment by creating an analogous
+`infrastructure-live/production/us-east-2/prod/vpc-app/terragrunt.hcl` file and running `terragrunt apply` in
+`infrastructure-live/production/us-east-2/prod/vpc-app/`. If you have multiple modules and you want to deploy all of
+them, you can use `terragrunt apply-all`. For example, to deploy _everything_ in the production account, you would
+do the following:
+
+----
+cd infrastructure-live/production
+terragrunt apply-all
+----
+
+*Benefits of this approach*
+
+* All your code is versioned. You can deploy different versions in different environments, and roll back to older
+  versions if necessary.
+* Backend configuration is DRY. No more copy/pasting values, no more manually setting `key` for each module.
+* CLI arguments are DRY. No need for long commands or extra wrapper scripts.
+* The full structure of all accounts, environments, and regions is visible at a glance in the `infrastructure-live`
+  repo.
+* Completely free and open source.
+* Deploy multiple modules using `apply-all`.
+
+*Drawbacks to this approach*
+
+* You have to install, learn, and manage a new tool / abstraction layer.
+* The scratch directory can make debugging/troubleshooting tricky.
+
+==== Updating
+
+Now that you have your Terraform module deployed, you can pull in updates as follows:
+
+. Subscribe to the monthly https://blog.gruntwork.io/tagged/gruntwork-newsletter[Gruntwork Newsletter] to be notified
+  of all updates to the Gruntwork Service Catalog.
```

Comment:
Alternatively they can also watch the repos relevant to them

Reply:
Updated: https://github.com/gruntwork-io/gruntwork-io.github.io/pull/151/commits/4603b48a336569db90352172790bc29347291c75